### ISSUE TITLE
[latching relay] adding support for one trigger pin instead of two

### ISSUE THREAD

**Issue by Gifford47** [arendst/Tasmota#18386]
### PROBLEM DESCRIPTION
Hey! I have this latching [relay](https://amzn.eu/d/iqW41fq)

I have tasmota 12.4 and configured the esp pin as relay_bi because the relay is triggered by low level. But every time I need 2 clicks to switch the relay. So I need 4 clicks to switch the relay on and off again.  Normally I only need 2 impulses to switch it on and off.
i think [this topic](https://github.com/arendst/Tasmota/issues/58) here is the first to introduce the implementation of a bistable relay.
like @arendst [said](https://github.com/arendst/Tasmota/issues/58#issuecomment-1270224511), i need to configure two configure two pins to switch the relay. My one has only one trigger pin to toggle both states. And there are many other bistable relays that have only one trigger pin.

It would be awesome if there is an option (i.e. setoption or a #define to compile my own build) to toggle such a relay with only one gpio. I know that there is no acknowledgment that the relay is in the right state, but as long as it is not accidentally switched in the boot phase or through incorrect configuration, you don't need it.

### REQUESTED INFORMATION
_Make sure your have performed every step and checked the applicable boxes before submitting your issue. Thank you!_

- [x] Read the [Contributing Guide and Policy](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md) and [the Code of Conduct](https://github.com/arendst/Tasmota/blob/development/CODE_OF_CONDUCT.md)
- [x] Searched the problem in [issues](https://github.com/arendst/Tasmota/issues)
- [x] Searched the problem in [discussions](https://github.com/arendst/Tasmota/discussions)
- [x] Searched the problem in the [docs](https://tasmota.github.io/docs/FAQ)
- [x] Searched the problem in the [chat](https://discord.gg/Ks2Kzd4)
- [ ] Device used (e.g., Sonoff Basic): _____
- [ ] Tasmota binary firmware version number used: _____
  - [ ] Pre-compiled
  - [ ] Self-compiled
- [ ] Flashing tools used: _____
- [ ] Provide the output of command: `Backlog Template; Module; GPIO 255`:
```lua
  Configuration output here:


```
- [ ] If using rules, provide the output of this command: `Backlog Rule1; Rule2; Rule3`:
```lua
  Rules output here:


```
- [ ] Provide the output of this command: `Status 0`:
```lua
  STATUS 0 output here:


```
- [ ] Set `weblog` to 4 and then, when you experience your issue, provide the output of the Console log:
```lua
  Console output here:


```

### EXPECTED BEHAVIOUR
The ESP is able to toggle a bistable relay (like the device above) with only one gpio (relay_b).

The procedure would be:
Init:
- define one gpio to relay_bi (or relay_b)
- gpio is at high level
->relay is at state 0

toggling relay:
- tasmota switches gpio to gnd for pulsetime(x) seconds
- tasmota switches gpio back to high level
-> relay is now in state 1

toggling relay to state 0:
- same procedure like above

### ADDITIONAL CONTEXT

**(Please, remember to close the issue when the problem has been addressed)**

**Comment by barbudor** [comment]
Use a standard Relay with Pulsetime to generate the toggling change
With a single pin bi-stable relay you will never be able to synchronize Tasmota power state with the relay state. If you need so, you need a feedback from the relay to a switch input

**Comment by Gifford47** [comment]
> Use a standard Relay with Pulsetime to generate the toggling change With a single pin bi-stable relay you will never be able to synchronize Tasmota power state with the relay state. If you need so, you need a feedback from the relay to a switch input

Why should it change without tasmota noticing? the only trigger comes from the tasmota gpio. i've had it running with custom firmware for weeks. but i'd like to run it with more features with tasmota.
[here ](https://github.com/arendst/Tasmota/commit/6031e5df98936e84e5547d4fb69421dde616ffcd)is the commit from the past

**Comment by arendst** [comment]
As @barbudor stated use pulsetime to toggle your relay. Why all the hassle for wanting to know a state where the relay itself not even knows it's state.

The current implementation (and the legacy implementation) both use two pins to know a fixed state of the relay. They are the known commercial bi-stable relay solutions too.

**Comment by barbudor** [comment]
There are still possible reasons for it to go wrong but that's your choice

With an esp8266 I would go by defining a virtual relay on an unused gpio and a rule set to trigger the actual control pin as a relay with pulsetile
Use WebUI and MQTT only to control the virtual relay and let the rules handle the real thing


If you have an esp32 you could most probably write a driver in berry

Making changes in Tasmota that won't be merged is not a good solution as it will require you to maintain this fork on your own and report all changes from the core to get the new features

**Comment by Gifford47** [comment]
> With an esp8266 I would go by defining a virtual relay on an unused gpio and a rule set to trigger the actual control pin as a relay with pulsetile Use WebUI and MQTT only to control the virtual relay and let the rules handle the real thing

Thanks (also for your time)! that´s a good idea! I will try that soon!

**Comment by Gifford47** [comment]
ok, for my case @barbudor has solved my issue. I toggle the relay with a rule watching var1. on both sides i enabled qos2 via mqtt.
i would be happy if such a function would be implemented anyway. But I can also understand if the benefit is too small for others.

**Comment by Gifford47** [comment]
Thanks!!!

**Comment by arendst** [comment]
Just had a look at this kind of relay modules and must conclude they are NOT real bi-stable relays. They are "normal" relays connected after a flip-flop; when turned on and power is lost they will come back turned off instead of turned on like a real bi-stable relay.

The only "bi-stable" feature they have is a single input allowing to toggle the relay by a low pulse of the flip-flop.

The current tasmota single-pin implementation stores the last power state and remembers the state after power on. This is probably NOT the state of the fake bi-stable relay after power on.

In short, it's a toy needing a different implemtation then the current one.....

**Comment by arendst** [comment]
OK. Fixed in latest dev branch. Single pin latching relay is NOT a two-pin bistable relay.
